### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by PiotrProkop**

### CODE REVIEW

**Comment by PiotrProkop** [comment]
@Quentin-M review please :smiley:

File: dependencies/config/config.go [hunk 1]
```diff
@@ -33,60 +34,59 @@ func init() {
 }
 
 func NewConfig(name string) Config {
-	var config Config
```
Comment (DTadrzak) [inline dependencies/config/config.go]: This function should return error.

File: dependencies/config/config.go [hunk 2]
```diff
@@ -33,60 +34,59 @@ func init() {
...
+			IP_ERLANG: strings.Replace(ip, ".", ",", -1),
+			HOSTNAME:  hostname},
+	}
 }
 
 func (c Config) IsResolved(entrypoint *entry.Entrypoint) (bool, error) {
-	logger.Info.Print(c.GetName())
-	err := CreateDirectory(c.GetName())
+	//Create directory to ensure it exists
+	err := createDirectory(c.GetName())
```
Comment (DTadrzak) [inline dependencies/config/config.go]: `if err := createDirectory(c.GetName()); err != nil {}`

File: dependencies/config/config.go [hunk 3]
```diff
@@ -33,60 +34,59 @@ func init() {
...
 	if err != nil {
 		return false, fmt.Errorf("Couldn't create directory: %v", err)
 	}
-	config, err := os.Create(c.GetName())
+	err = createAndTemplateConfig(c.GetName(), c.params)
```
Comment (DTadrzak) [inline dependencies/config/config.go]: ditto

File: dependencies/config/config.go [hunk 4]
```diff
@@ -33,60 +34,59 @@ func init() {
...
-		return false, err
+		return false, fmt.Errorf("Cannot template %s: %v", c.GetName(), err)
 	}
 	return true, nil
 
 }
 
+func createAndTemplateConfig(name string, params configParams) (err error) {
+	config, err := os.Create(name)
+	if err != nil {
```
Comment (DTadrzak) [inline dependencies/config/config.go]: `if err != nil { return }`

File: dependencies/container/container.go [hunk 5]
```diff
@@ -2,7 +2,7 @@ package container
 
 import (
 	"fmt"
```
Comment (DTadrzak) [inline dependencies/container/container.go]: please split imports into two groups built-in and external packages

File: dependencies/daemonset/daemonset.go [hunk 6]
```diff
@@ -19,12 +19,12 @@ func init() {
...
 	if daemonsetsDeps := env.SplitEnvToList(daemonsetEnv); daemonsetsDeps != nil {
 		for _, dep := range daemonsetsDeps {
-			entry.Register(NewDaemonset(dep))
+			entry.Register(NewDaemonSet(dep))
 		}
 	}
 }
 
-func NewDaemonset(name string) Daemonset {
+func NewDaemonSet(name string) Daemonset {
```
Comment (DTadrzak) [inline dependencies/daemonset/daemonset.go]: you should also change the type name to DaemonSet

File: entrypoint/entrypoint.go [hunk 7]
```diff
@@ -12,18 +12,19 @@ import (
...
 var dependencies []Resolver // List containing all dependencies to be resolved
 const (
-	DependencyPrefix = "DEPENDENCY_" //Prefix for env variables
+	//DependencyPrefix is a prefix for env variables
+	DependencyPrefix = "DEPENDENCY_"
 	interval         = 2
 )
 
-// Object containing k8s client
+// Entrypoint is a main struct which check dependencies
```
Comment (DTadrzak) [inline entrypoint/entrypoint.go]: s/check/checks/

File: entrypoint/entrypoint.go [hunk 8]
```diff
@@ -38,6 +39,7 @@ func NewEntrypoint(client *cl.Client) (entry *Entrypoint, err error) {
 	return entry, err
 }
 
+//Resolve is a main loop whic iterates through all dependencies and resolves them
```
Comment (DTadrzak) [inline entrypoint/entrypoint.go]: s/whic/which/

File: entrypoint/entrypoint.go [hunk 9]
```diff
@@ -61,12 +63,13 @@ func (e *Entrypoint) Resolve() {
...
 }
 
+//Resolver is an interface which all dependencies should implement
 type Resolver interface {
-	//	GetType() string
 	IsResolved(entrypoint *Entrypoint) (bool, error)
 	GetName() string
 }
 
+//Register is a function which register new dependencies
```
Comment (DTadrzak) [inline entrypoint/entrypoint.go]: s/register/registers/

File: kubernetes-entrypoint.go [hunk 10]
```diff
@@ -20,18 +21,22 @@ import (
...
 	}
 	entrypoint.Resolve()
 
-	if command = env.SplitEnvToList("COMMAND", " "); len(command) == 0 {
+	if comm = env.SplitEnvToList("COMMAND", " "); len(comm) == 0 {
 		logger.Error.Printf("COMMAND env is empty")
 		os.Exit(1)
 	}
-	comm.ExecuteCommand(command)
+	err = command.Execute(comm)
```
Comment (DTadrzak) [inline kubernetes-entrypoint.go]: `if err = command.Execute(comm); err != nil {}`

File: entrypoint/entrypoint.go (hunk 7 above)
Comment (DTadrzak) [inline entrypoint/entrypoint.go]: s/check/checks

File: entrypoint/entrypoint.go (hunk 8 above)
Comment (DTadrzak) [inline entrypoint/entrypoint.go]: s/whic/which/

File: entrypoint/entrypoint.go (hunk 9 above)
Comment (DTadrzak) [inline entrypoint/entrypoint.go]: s/register/registers/

File: util/util.go [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strings"
+)
+
+func GetIp() (ip string, err error) {
+	iface := os.Getenv("INTERFACE_NAME")
```
Comment (DTadrzak) [inline util/util.go]: `if iface := os.Getenv("INTERFACE_NAME"); iface == "" {}`

File: util/util.go [hunk 12]
```diff
@@ -0,0 +1,27 @@
+package util
```
Comment (DTadrzak) [inline util/util.go]: You can also move function get_hostname here.

**Comment by Quentin-M** [comment]
I'll review today.

File: mocks/daemonset.go [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+)
+
+type dClient struct {
+}
+
+func (d dClient) Get(name string) (*extensions.DaemonSet, error) {
+	if name != "lgtm" {
+		return nil, fmt.Errorf("Mock daemonset didnt work")
+	}
+	ds := new(extensions.DaemonSet)
```
Comment (Quentin-M) [inline mocks/daemonset.go]:
```
ds := &extensions.DaemonSet{
  Name: name,
  Spec: DaemonSetSpec{
    Selector: &unv.LabelSelector{
      MatchLabels: map[string]string{"name": "test"}
    },
  },
}
```

File: mocks/endpoints.go [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+)
+
+type eClient struct {
+}
+
+func (e eClient) Get(name string) (*api.Endpoints, error) {
+	if name != "lgtm" {
+		return nil, fmt.Errorf("Mock endpoint didnt work")
+	}
+	endpoint := new(api.Endpoints)
```
Comment (Quentin-M) [inline mocks/endpoints.go]: Same as above, everywhere like this.

**Pull request merged**